File: native/src/main/java/com/example/app/ui/components/CustomMediumText.kt
package com.example.app.ui.components

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.sp

@Composable
fun CustomMediumText(
    text: String,
    size: TextUnit = 11.sp,
    lineHeight: TextUnit = 23.sp,
    color: Color = Color.Black
) {
    Text(
        text = text,
        style = TextStyle(
            color = color,
            fontWeight = FontWeight.Medium,
            fontSize = size,
            lineHeight = lineHeight,
        )
    )
}

File: non_native/lib/view/components/custom_medium_text.dart
import 'package:flutter/material.dart';

class CustomMediumText extends StatelessWidget {
  final String text;
  final double size;
  final double lineHeight;
  final Color color;

  const CustomMediumText({
    Key? key,
    required this.text,
    this.size = 11.0,
    this.lineHeight = 23.0,
    this.color = Colors.black,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Text(
      text,
      style: TextStyle(
        color: color,
        fontWeight: FontWeight.w500,
        fontSize: size,
        height: lineHeight / size,
      ),
    );
  }
}
